app/controllers/opendata/agents/nodes/app/app_controller.rb

```diff
@@ -11,159 +11,159 @@ class Opendata::Agents::Nodes::App::AppController < ApplicationController
-
-    def show_executed
-      @cur_node.layout = nil
-      # @app = Opendata::App.site(@cur_site).find(params[:app])
-      @add = false
-      if params[:tab_display] == "tab_html"
-        @add = true
+      @mode = :cancel if point = Opendata::AppPoint.where(cond).first
+    end
+  end
+
+  def add_point
+    @cur_node.layout = nil
+    raise "403" unless logged_in?(redirect: false)
+
+    cond = { site_id: @cur_site.id, member_id: @cur_member.id, app_id: @app.id }
+
+    if point = Opendata::AppPoint.where(cond).first
+      point.destroy
+      @app.inc point: -1
+      @mode = :add
+    else
+      Opendata::AppPoint.new(cond).save
+      @app.inc point: 1
+      @mode = :cancel
+    end
+
+    render :show_point
+  end
+
+  def point_members
+    @cur_node.layout = nil
+    @items = Opendata::AppPoint.where(site_id: @cur_site.id, app_id: @app.id)
+  end
+
+  def show_ideas
+    @cur_node.layout = nil
+    @login_mode = logged_in?(redirect: false)
+    @idea_comment = Opendata::IdeaComment
+  end
+
+  def show_executed
+    @cur_node.layout = nil
+    # @app = Opendata::App.site(@cur_site).find(params[:app])
+    @add = false
+    if params[:tab_display] == "tab_html"
+      @add = true
+    end
+    render
+  end
+
+  def add_executed
+    @cur_node.layout = nil
+    @add = false
+    if @app.present?
+      exec = @app.executed.to_i
+      @app.executed = exec + 1
+      res = @app.save(validate: false)
+      if !res
+        @app.executed = exec
       end
-      render
     end
+    render :show_executed
+  end
 
-    def add_executed
-      @cur_node.layout = nil
-      @add = false
-      if @app.present?
-        exec = @app.executed.to_i
-        @app.executed = exec + 1
-        res = @app.save(validate: false)
-        if !res
-          @app.executed = exec
-        end
-      end
-      render :show_executed
-    end
-
-    def full
-      @cur_node.layout = nil
-      @item = @app
-      @app_html = @item.appfiles.where(filename: "index.html").first
-    end
+  def full
+    @cur_node.layout = nil
+    @item = @app
+    @app_html = @item.appfiles.where(filename: "index.html").first
+  end
 
-    def app_index
-      @cur_node.layout = nil
-      if @appfile.present?
-        send_file @appfile.file.path, type: @appfile.content_type, filename: @appfile.filename,
-          disposition: :inline, x_sendfile: true
-      end
+  def app_index
+    @cur_node.layout = nil
+    if @appfile.present?
+      send_file @appfile.file.path, type: @appfile.content_type, filename: @appfile.filename,
+        disposition: :inline, x_sendfile: true
     end
+  end
 
-    def text
-      @cur_node.layout = nil
-      if @appfile.present?
-        send_file @appfile.file.path, :type => "text/plain", filename: @appfile.filename,
-          disposition: :inline, x_sendfile: true
-      end
+  def text
+    @cur_node.layout = nil
+    if @appfile.present?
+      send_file @appfile.file.path, :type => "text/plain", filename: @appfile.filename,
```

Comment:
__[brakeman]__
* Parameter value used in file name [:ref](http://brakemanscanner.org/docs/warning_types/file_access/)

<!-- Comment by [SideCI](https://sideci.com) -->